lib/taskana-core/src/main/java/pro/taskana/ClassificationService.java

```diff
@@ -44,6 +46,34 @@
      */
     Classification getClassification(String key, String domain) throws ClassificationNotFoundException;
 
+    /**
+     * Delete a non-root-classification.
+     *
+     * @param classificationKey
+     *              the key of the classification you want to delete.
+     * @param domains
+     *              the domains for which you want to delete the classification.
+     * @throws ClassificationInUseException
+     *              if there are Task existing, which refer to this classification.
+     * @throws NotAuthorizedException
+     *              if you're not authorized to delete the classification.
+     * @throws ClassificationNotFoundException
+     *              if for an domain no classification specification is found.
+     */
+    void deleteClassification(String classificationKey, String... domains) throws ClassificationInUseException, NotAuthorizedException, ClassificationNotFoundException;
+
+    /**
+     * Delete a root-classification with all children.
+     *
+     * @param classificationKey
+     *              the key of the classification you want to delete.
+     * @throws ClassificationInUseException
+     *              if there are Task existing, which refer to this classification.
+     * @throws NotAuthorizedException
+     *              if you're not authorized to delete the classification.
+     */
+    void deleteRootClassification(String classificationKey) throws ClassificationInUseException, NotAuthorizedException;
```

Comment:
Have a look at the comment above. (line 63)

Reply:
It is not the root classification. It is the classification from the "master domain" (domain == ""). Maybe we don't need a seperate method?